project/MimaSettings.scala

```diff
@@ -19,7 +19,9 @@ object MimaSettings {
         exclude[IncompatibleMethTypeProblem]("zio.Config#*"),
         exclude[DirectMissingMethodProblem]("zio.Config#*"),
         exclude[DirectMissingMethodProblem]("zio.Config.*"),
-        exclude[MissingTypesProblem]("zio.Config$Secret$")
+        exclude[MissingTypesProblem]("zio.Config$Secret$"),
+        exclude[DirectMissingMethodProblem]("zio.metrics.MetricKey.copy"),
```

Comment:
I don't know if we can just ignore binary compatibility here.

Reply:
MetricKey has private constructor, should be safe to ignore

Reply:
Those methods are still public so I don't think that is correct.

```scala
val metricKey = MetricKey.counter("foo")
metricKey.copy(name = "bar")
```

Reply:
We can get around this by changing `MetricKey` from a `final case class` to a `sealed case class`, making `desription` a `def`, and overriding `copy` to copy the description.

Reply:
```scala
sealed case class MetricKey[+Type] private (
  name: String,
  keyType: Type,
  tags: Set[MetricLabel] = Set.empty
) { self =>

  def description: Option[String] =
    None

  def copy[Type2](name: String = name, keyType: Type2 = keyType, tags: Set[MetricLabel] = tags): MetricKey[Type2] =
    new MetricKey(name, keyType, tags) {
      override def description: Option[String] =
        self.description
    }
}
```

Reply:
Done. Thank you for great idea!